### PR TITLE
[NFC] [SYCL] Improve debugability of 'cache build result' test

### PR DESCRIPTION
**Pull request by s-kanaev**
The test may not be reproduced locally and it's not clear from the log what issue is encountered

### CODE REVIEW

File: SYCL/KernelAndProgram/cache-build-result.cpp [hunk 1]
```diff
@@ -37,6 +37,11 @@ void test() {
         assert(Result == e.get_cl_code() && "Exception code differs");
       }
     } catch (...) {
+      try {
+        std::rethrow_exception(std::current_exception());
+      } catch (std::exception &E) {
+        fprintf(stderr, "Unexpected exception: %s\n", E.what());
+      }
       assert(false && "There must be cl::sycl::compile_program_error");
     }
```
Comment (romanovvlad) [inline SYCL/KernelAndProgram/cache-build-result.cpp]:
```suggestion
    } catch (std::exception &E) {
      fprintf(stderr, "Unexpected exception: %s\n", E.what());
    } catch (...) {
      assert(false && "There must be cl::sycl::compile_program_error");
    }
```
?

Comment (vladimirlaz) [reply SYCL/KernelAndProgram/cache-build-result.cpp]: We will never go to line 41-43 block. The assert may be moved to line 39-40 block

Comment (romanovvlad) [reply SYCL/KernelAndProgram/cache-build-result.cpp]: >> We will never go to line 41-43 block. 

Could you please clarify why?

Comment (vladimirlaz) [reply SYCL/KernelAndProgram/cache-build-result.cpp]: sorry. my fault. I was thinking that all we throw is inherited from std::exception. It is definitely incorrect. So ignore my comment

**Pull request closed without merging**